src/main/kotlin/net/atos/zac/signalering/SignaleringenJob.kt

```diff
@@ -0,0 +1,377 @@
+import net.atos.zac.gebruikersvoorkeuren.model.TabelInstellingen
+import net.atos.zac.signalering.model.Signalering
+import net.atos.zac.signalering.model.SignaleringDetail
+import net.atos.zac.signalering.model.SignaleringTarget
+import net.atos.zac.signalering.model.SignaleringType
+import net.atos.zac.signalering.model.SignaleringVerzendInfo
+import net.atos.zac.signalering.model.SignaleringVerzondenZoekParameters
+import net.atos.zac.util.UriUtil
+import net.atos.zac.zaaksturing.ZaakafhandelParameterService
+import net.atos.zac.zoeken.ZoekenService
+import net.atos.zac.zoeken.model.DatumRange
+import net.atos.zac.zoeken.model.DatumVeld
+import net.atos.zac.zoeken.model.FilterVeld
+import net.atos.zac.zoeken.model.FilterWaarde
+import net.atos.zac.zoeken.model.ZoekObject
+import net.atos.zac.zoeken.model.ZoekParameters
+import net.atos.zac.zoeken.model.index.ZoekObjectType
+import net.atos.zac.zoeken.model.zoekobject.ZaakZoekObject
+import nl.lifely.zac.util.AllOpen
+import nl.lifely.zac.util.NoArgConstructor
+import org.flowable.task.api.Task
+import java.time.LocalDate
+import java.util.Objects
+import java.util.UUID
+import java.util.function.Consumer
+import java.util.logging.Logger
+
+@ApplicationScoped
+@Transactional
+@Suppress("TooManyFunctions")
+@NoArgConstructor
+@AllOpen
+class SignaleringenJob @Inject constructor(
+    private val signaleringenService: SignaleringenService,
+    private val configuratieService: ConfiguratieService,
+    private val ztcClientService: ZTCClientService,
+    private val zaakafhandelParameterService: ZaakafhandelParameterService,
+    private val zoekenService: ZoekenService,
+    private val takenService: TakenService
+) {
+
+    companion object {
+        private val LOG: Logger = Logger.getLogger(SignaleringenJob::class.java.name)
+        const val ZAAK_AFGEHANDELD_QUERY: String = "zaak_afgehandeld"
+    }
+
+    fun signaleringenVerzenden() {
+        zaakSignaleringenVerzenden()
+        taakSignaleringenVerzenden()
+    }
+
+    /**
+     * This is the batchjob to send E-Mail warnings about cases that are approaching their einddatum gepland or their
+     * uiterlijke einddatum afdoening.
+     */
+    fun zaakSignaleringenVerzenden() {
+        val info = SignaleringVerzendInfo()
+        LOG.info("Zaak signaleringen verzenden: gestart...")
+        ztcClientService.listZaaktypen(configuratieService.readDefaultCatalogusURI())
+            .forEach(
+                Consumer<ZaakType> { zaaktype: ZaakType ->
+                    val parameters = zaakafhandelParameterService.readZaakafhandelParameters(
+                        URIUtil.parseUUIDFromResourceURI(zaaktype.url)
+                    )
+                    if (parameters.einddatumGeplandWaarschuwing != null) {
+                        info.streefdatumVerzonden += zaakEinddatumGeplandVerzenden(
+                            zaaktype,
+                            parameters.einddatumGeplandWaarschuwing
```

Comment:
```suggestion
                            it
```

(with the previous suggestion)